src/HDF5.jl

```diff
@@ -1371,10 +1373,10 @@ function read{S<:String}(obj::DatasetOrAttribute, ::Type{Array{S}})
         ret = Array{S}(sz)
         if isvar
             # Variable-length
-            buf = Vector{Ptr{UInt8}}(len)
+            buf = Vector{Cstring}(len)
             h5t_set_size(memtype_id, H5T_VARIABLE)
             readarray(obj, memtype_id, buf)
-            # FIXME? Who owns the memory for each string? Will Julia free it?
+            # FIXME? Who owns the memory for each string? Julia v0.6+ won't free it.
```

Comment:
I only answered one of the questions there. I suspect that implies the answer the the first question (is that this is a memory leak and we should be manually freeing the memory here after the call to `unsafe_string`), but perhaps instead HDF5 owns the memory (as a pointer into the mmap'd file?) and attempting to free it would immediately cause it to crash.